packages/devextreme/js/__internal/ui/date_range_box/m_date_range_box.ts

```diff
@@ -41,1003 +47,1081 @@ const DROP_DOWN_EDITOR_ACTIVE_CLASS = 'dx-dropdowneditor-active';
-            this._createOpenAction();
-        }
-        this._openAction();
-    }
-
-    _createCloseAction() {
-        this._closeAction = this._createActionByOption('onClosed', {
-            excludeValidators: ['disabled', 'readOnly']
-        });
-    }
-
-    _raiseCloseAction() {
-        if(!this._closeAction) {
-            this._createCloseAction();
-        }
-        this._closeAction();
-    }
-
-    _createEventAction(eventName) {
-        this[`_${camelize(eventName)}Action`] = this._createActionByOption(`on${eventName}`, {
-            excludeValidators: ['readOnly']
-        });
-    }
-
-    _raiseAction(eventName, event) {
-        const action = this[`_${camelize(eventName)}Action`];
-        if(!action) {
-            this._createEventAction(eventName);
-        }
-        this[`_${camelize(eventName)}Action`]({ event });
-    }
-
-    _initTemplates() {
-        this._templateManager.addDefaultTemplates({
-            dropDownButton: new FunctionTemplate(function(options) {
-                const $icon = $('<div>').addClass(DROP_DOWN_EDITOR_BUTTON_ICON);
-                $(options.container).append($icon);
-            })
-        });
-        this.callBase();
-    }
+    // @ts-expect-error
+    this._startDateBox = this._createComponent(this._$startDateBox, MultiselectDateBox, this._getStartDateBoxConfig());
+    this._startDateBox.NAME = '_StartDateBox';
+  }
 
-    _getDefaultButtons() {
-        return [
-            { name: 'clear', Ctor: ClearButton },
-            { name: 'dropDown', Ctor: DropDownButton }
-        ];
-    }
+  _renderEndDateBox() {
+    this._$endDateBox = $('<div>')
+      .addClass(END_DATEBOX_CLASS)
+      .appendTo(this.$element());
+    // @ts-expect-error
+    this._endDateBox = this._createComponent(this._$endDateBox, MultiselectDateBox, this._getEndDateBoxConfig());
+    this._endDateBox.NAME = '_EndDateBox';
+  }
 
-    _initMarkup() {
-        this.$element()
-            .addClass(DATERANGEBOX_CLASS)
-            .addClass(TEXTEDITOR_CLASS)
-            .addClass(DROP_DOWN_EDITOR_CLASS);
+  _renderSeparator() {
+    const $icon = getImageContainer(SEPARATOR_ICON_NAME);
+    this._$separator = $('<div>')
+      .addClass(DATERANGEBOX_SEPARATOR_CLASS)
+      .prependTo(this.$element());
 
-        this._toggleDropDownEditorActiveClass();
-        this._toggleEditorLabelClass();
+    this._renderPreventBlurOnSeparatorClick();
 
-        this._toggleReadOnlyState();
-        this._renderStylingMode();
-        // TODO: probably it need to update styling mode for dropDown in buttons container. It depends from design decision
+    $icon?.appendTo(this._$separator);
+  }
 
-        this._renderEndDateBox();
-        this._renderSeparator();
-        this._renderStartDateBox();
+  _renderPreventBlurOnSeparatorClick(): void {
+    // @ts-expect-error
+    const eventName = addNamespace('mousedown', this.NAME);
 
+    eventsEngine.off(this._$separator, eventName, undefined);
```

Comment:
Are you sure it's safe? if `.off` checks inside of it for `arguments.legtht`, it will break
I think it's better to change `.d.ts`